scripts/gulp/collect-files.js

```diff
@@ -0,0 +1,16 @@
+const through2 = require("through2");
+
+const collectedFiles = {};
+
+function collector(namespace) {
+  const namedCollectedFiles = (collectedFiles[namespace] = []);
+
+  return through2.obj(function (file, _, cb) {
+    namedCollectedFiles.push(file);
+    cb(null, file);
+  });
+}
+
+collector.get = (namespace) => collectedFiles[namespace];
```

Comment:
Adding a function to a function is not a good practice. JS is too flexible! Better to export both functions

Reply:
That's good advice. I mimicked the API from some other library that does something similar, but perhaps that's not necessary.